sycl/doc/extensions/DeviceGlobal/SYCL_INTEL_device_global.asciidoc

```diff
@@ -398,6 +395,214 @@ Available only if `sycl::is_property_of_v<propertyT, sycl::ext::oneapi::device_g
 
 |===
 
+=== Properties for device global variables
+
+The `device_global` class supports several compile-time-constant properties.
+If specified, these properties are included in the `PropertyListT` template
+parameter as shown in this example:
+
+[source,c++]
+----
+using namespace sycl::ext::oneapi;
+
+device_global<MyClass, property_list_t<device_image_life::value_t>> dm1;
+device_global<int[4], property_list_t<copy_access::value_t<copy_access::access::read>> dm2;
+----
+
+The following code synopsis shows the set of supported properties, and the
+following table describes their effect.
+
+[source,c++]
+----
+namespace sycl::ext::oneapi {
+
+struct device_image_life {
+  using value_t = property_value<device_image_life>;
+};
+
+struct copy_access {
+  enum class access: /*unspecified*/ {
+    read,
+    write,
+    read_write,
+    no_access
+  };
+  template<access A>
+  using value_t = property_value<copy_access, std::integral_constant<access, A>>;
+
+struct init_via {
+  enum class trigger: /*unspecified*/ {
+    reprogram,
+    reset
+  };
+  template<trigger T>
+  using value_t = property_value<init_via, std::integral_constant<trigger, T>>;
+};
+
+struct implement_in_csr {
+  template <bool Enable>
+  using value_t = property_value<implement_in_csr, std::bool_constant<Enable>>;
+};
+
+
+inline constexpr device_image_life::value_t device_image_life_v;
+
+template<copy_access::access A>
+inline constexpr copy_access::value_t<A> copy_access_v;
+
+template<init_via::trigger T>
+inline constexpr init_via::value_t<T> init_via_v;
+
+template<bool Enable>
+inline constexpr implement_in_csr::value_t<Enable> implement_in_csr_v;
+
+} // namespace sycl::ext::oneapi
+----
+
+[frame="topbot",options="header"]
+|===
+|Property |Description
+
+a|
+[source,c++]
+----
+device_image_life
```

Comment:
Upon initial reading I assumed that "life" here was a short version of "lifetime" (as in "object lifetime").  Would something like `per_device_image` or `allocate_per_device_image` be clearer here, do you think?

Or perhaps something extensible like a `visibility` property with an enum, where initially the only two options are `device` and `device_image`?

Reply:
I think sycl::ext::oneapi is already a lot of typing so I'd rather we not add the extra nesting level.